docs/architecture/agents.md

```diff
@@ -42,6 +42,15 @@ More specifically, an application agent requires the following resources to exis
 * A Service Account for the agent
 * A RoleBinding that binds the ServiceAccount to the Role
 
+When a ServiceBinding is created on an application namespace, the Application Agent looks for resources mentioned in its specification.
+
+Primaza Application Agent runs a dynamic informer for `Application` Resources mentioned in ServiceBinding specification.
+
+The informer monitors changes to `Application` matching the Service Binding specifications and updates the Service Binding accordingly.
+
+* If the `Application` Resource mentioned in Service Binding specification is updated or created, the secrets referenced by ServiceBinding resources will be projected into all  the  application compute resources (CronJobs, Deployments, ReplicaSets, DaemonSets etc…) that contain the labels matching the label selector or the specific workload if specified by name.
```

Comment:
is this the intended behavior? I'd say we should enforce Application's `APIVersion` and `Kind` to be defined